openedx/core/djangoapps/user_authn/views/logout.py

```diff
@@ -59,7 +60,12 @@ def dispatch(self, request, *args, **kwargs):
 
         # If we are using studio logout directly and there is not OIDC logouts we can just redirect the user
         if settings.FEATURES.get('DISABLE_STUDIO_SSO_OVER_LMS', False) and not self.oauth_client_ids:
-            response = redirect(self.target)
+            # If ALLOW_SAML_LOGOUT_FROM_URL is set to True, we need to go to the logout page instead of redirect,
```

Comment:
What if we try to reduce "Cyclomatic Complexity" using just one if else expression:
```suggestion
if settings.FEATURES.get('DISABLE_STUDIO_SSO_OVER_LMS', False) and not self.oauth_client_ids and not configuration_helpers.get_value('ALLOW_SAML_LOGOUT_FROM_URL', False):
    response = redirect(self.target)
else:
    response = super(LogoutView, self).dispatch(request, *args, **kwargs)
            # If ALLOW_SAML_LOGOUT_FROM_URL is set to True, we need to go to the logout page instead of redirect,
```